<launch>
    <arg name="topic_in" default="points"/>
    <arg name="topic_out" default="tf_cloud"/>
    <arg name="from_frame" default="foo_link"/>
    <arg name="to_frame" default="bar_link"/>

    <arg name="point_x" default="1.0"/>
    <arg name="point_y" default="1.0"/>
    <arg name="point_z" default="1.0"/>

    <node pkg="transform_pointcloud_srv" exec="transform_pointcloud_srv_node" name="transform_pointcloud_srv_node" />

    <node pkg="tf2_ros" exec="static_transform_publisher" name="static_transform_publisher" 
        args="0 0 1.0 0 0 0 $(var to_frame) $(var from_frame)" />

    <node pkg="transform_pointcloud_srv" exec="test_talker.py" name="test_talker">
        <param name="topic_name" value="$(var topic_in)"/>
        <param name="reference_frame" value="$(var from_frame)"/>
        <param name="point_x" value="$(var point_x)"/>
        <param name="point_y" value="$(var point_y)"/>
        <param name="point_z" value="$(var point_z)"/>
    </node>

    <node pkg="transform_pointcloud_srv"  exec="test_client.py" name="test_client">
        <param name="topic_name" value="$(var topic_in)"/>
    </node>
</launch>